pkgs/top-level/all-packages.nix

```
@@ -16755,6 +16755,7 @@ with pkgs;
   php80Extensions = recurseIntoAttrs php80.extensions;
   php80Packages = recurseIntoAttrs php80.packages;
 
+  phpactor = callPackage ../development/tools/phpactor { composer = phpPackages.composer; };
```

```suggestion
  phpactor = callPackage ../development/tools/phpactor {  inherit (__splicedPackages.phpPackages) composer; };
```

`__splicedPackages` so cross works

---

Can you explain this? I've never seen so far this way of doing. What are the benefits?

---

if a package isn't gotten from __splicedPackages then the package won't have __spliced and the correct version(for build) can't be chosen when the package is in nativeBuildInputs. So cross compiling will be broken because the executables will be for host instead of for build.

https://github.com/NixOS/nixpkgs/issues/204303

---

Or maybe use `php.packages.composer` inside the derivation? That way you will hopefully not to deal with splicing explicitly and it will fit better with https://github.com/NixOS/nixpkgs/issues/204303.